src/main/java/com/firebolt/jdbc/connection/FireboltConnection.java

```diff
@@ -131,12 +132,20 @@ private void connect() throws FireboltException {
 			//When running packdb locally, the login properties are the session properties
 			sessionProperties = loginProperties;
 		}
+		assertDatabaseExisting(sessionProperties.getDatabase());
+
 		log.debug("Connection opened");
 	}
 
-	private FireboltProperties getSessionPropertiesForNonSystemEngine() throws FireboltException {
+	private FireboltProperties getSessionPropertiesForNonSystemEngine() throws SQLException {
 		Engine engine = fireboltEngineService.getEngine(loginProperties.getEngine(), loginProperties.getDatabase());
-		return loginProperties.toBuilder().host(engine.getEndpoint()).engine(engine.getName()).systemEngine(false).build();
+		return loginProperties.toBuilder().host(engine.getEndpoint()).engine(engine.getName()).systemEngine(false).database(engine.getDatabase()).build();
+	}
+
+	private void assertDatabaseExisting(String database) throws SQLException {
```

Comment:
Just a question, should't it also be `throws FireboltException`, since we do `throw new FireboltException(...)`?

Reply:
No. Generally the interfaces should use general terms. Prefer interface to class, prefer interface/class higher in hierarchy than specific one. This is the basics of the OOD. 
Specifically here. `FireboltException` extends `SQLException` and therefore can be thrown from any  function that is declared as `throws SQLException`. The opposite is incorrect. If your function is declared as `throws FireboltException` it cannot throw `SQLException`. This is the reason that there many  places in code that look like the following: